### PR TITLE
Update Sensor API for new-style event pages

### PR DESCRIPTION
**Pull request by Elchi3**
This PR updates the Sensor API for new-style event pages.

- Ran `yarn content move Web/API/Sensor/onerror Web/API/Sensor/error_event`
- Ran `yarn content move Web/API/Sensor/onreading Web/API/Sensor/reading_event`
- Ran `yarn content move Web/API/Sensor/onactivate Web/API/Sensor/activate_event`

- Updated https://developer.mozilla.org/en-US/docs/Web/API/Sensor:
  - Moved the "Event handlers" section to an "Events" section
  - Linked to xxx_event pages instead of onxxx
  - Sort events alphabetically. Update descriptions

- Updated `Web/API/Sensor/activate_event` (moved in the first step)
  - Updated to new-style event page. See https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_event_subpage_template for the template.
  - Added example code.

- Updated `Web/API/Sensor/error_event` (same update as activate event, but note this is of type `SensorErrorEvent`)
- Updated `Web/API/Sensor/reading_event` (same update as activate event)

- Searched all of mdn/content for "sensor.on" to update any pages that reference sensor "on" event handlers and updated them. Did some adjacent changes like adding inheritance diagrams and fixing syntax boxes.

BCD change is approved in https://github.com/mdn/browser-compat-data/pull/13926

### CODE REVIEW

**Comment by teoli2003** [comment]
The `Sensor` interface has an internal state machine that is not documented (See https://w3c.github.io/sensors/#sensor-lifecycle ). I think we should document it on the main interface (in non spec terminology). The three events notify changes in this state machine, so we should make it clearer.

Let me try to write this here in review comments.

File: files/en-us/web/api/ambientlightsensor/index.md [hunk 1]
```diff
@@ -31,17 +33,25 @@ If a feature policy blocks use of a feature it is because your code is inconsist
 - {{domxref('AmbientLightSensor.illuminance')}}
   - : Returns the current light level in [lux](https://en.wikipedia.org/wiki/Lux) of the ambient light level around the hosting device.
 
+## Methods
+
+_`AmbientLightSensor` doesn't have own methods. However, it inherits methods from its parent interfaces, {{domxref("Sensor")}} and {{domxref("EventTarget")}}._
+
+## Events
+
+_`AmbientLightSensor` doesn't have own events. However, it inherits events from its parent interface, {{domxref('Sensor')}}._
```
Comment (teoli2003) [inline files/en-us/web/api/ambientlightsensor/index.md:42]: This text sounds strange to me. I wonder if we should not speak about target here. Something like:
```suggestion
_`AmbientLightSensor` is not the target of specific events. However, it is the target of events aimed at its parent interface, {{domxref('Sensor')}}._
```

Not sure if it is good enough, though.

File: files/en-us/web/api/sensor/activate_event/index.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+title: 'Sensor: activate event'
+slug: Web/API/Sensor/activate_event
+tags:
+  - API
```
Comment (teoli2003) [inline files/en-us/web/api/sensor/activate_event/index.md:5]: Given most these tags are not used, shouldn't we just have 
- Reference
- Event
as tags (Event is used by sidebar I think, and reference is a kind of page type, that will allow us to grep reference pages at some point).
?

File: files/en-us/web/api/ambientlightsensor/index.md (hunk 1 above)
Comment (Elchi3) [reply files/en-us/web/api/ambientlightsensor/index.md:42]: Hm, these sentences are inconsistent throughout. Another version is 

"_No specific events; inherits events from its ancestor, {{domxref('Sensor')}}._"

Maybe that is better?

I would like to have something to use everywhere and maybe also to have it consistent with what is written for Properties and Methods.

File: files/en-us/web/api/sensor/activate_event/index.md (hunk 2 above)
Comment (Elchi3) [reply files/en-us/web/api/sensor/activate_event/index.md:5]: I think they don't hurt at the moment (they are copied from moving the page). I guess at some point we will have a better understanding how we want to use tags and either remove them in bulk or do something else with them.

**Review by teoli2003 — APPROVED** [review]
This looks ok to me. I don't know if you want to coordinate with the landing of the BCD PR or if this can go ahead now.

**Comment by teoli2003** [comment]
Discussed with Florian. We are landing this now to prevent rot.

**Pull request merged**